include/NAS2D/Renderer/Vector.h

```diff
@@ -39,6 +42,30 @@ struct Vector {
 		};
 	}
 
+	Vector& operator*=(BaseType scalar) {
+		x *= scalar;
+		y *= scalar;
+		return *this;
+	}
+	Vector& operator/=(BaseType scalar) {
+		if (scalar == 0) {
+			throw std::domain_error("Cannot divide vector by 0");
```

Comment:
This `if` check isn't needed and certainly not the custom throw. Throwing on divide by zero is built in.

Reply:
Actually, no, it's undefined behaviour. For integer operations pretty much anything can happen. For floating point, there is generally some hardware defined behaviour to produce infinity values.

----

Though I realize now in hindsight that I forgot to add checks for divide by 0 to the unit tests.